### PR TITLE
[HMA-UI] add eslint and prettier ci action for webapp

### PR DESCRIPTION
**Pull request by BarrettOlson**
Summary
---------

lint is love. This looks like a major refactor but the code logic shouldn't have changed much (added some props here and there for `eslint-plugin-jsx-a11y` etc and a few minor refactors to make eslint happy). 

Warning: you won't be able to `npm run build` (and `npm start` might be flaky) if you have eslint errors. If you _really_ need to build and don't have time to suppress the rule in the config of inline temporary you can use.
`DISABLE_ESLINT_PLUGIN=true npm run build`  VSCode plugins will be your friends as well.

I plan on updating `webapp/README.md` once I get some initial feedback.

Test Plan
---------
Tested that existing match results and photo upload still work. 
now we can run:
```
npx eslint src --ext .js,.jsx  
npx prettier --check src 
```

### CODE REVIEW

File: .github/workflows/hma-webapp-ci.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        run: yarn
+      - name: Run ESLint
+        run: eslint src
```
Comment (schatten) [inline .github/workflows/hma-webapp-ci.yaml:28]: Maybe related to the failure we are seeing, but perhaps `./node_modules/.bin/eslint` would be more appropriate.

**Review by eriksendc — DISMISSED** [review]

**Review by schatten — APPROVED** [review]
Approving to unblock. I'm seeing that the workflow can be tested while a PR. 

But this seems to be a Draft PR, so dunno why I'm even reviewing.

File: .github/workflows/hma-webapp-ci.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        run: npm install
+      - name: Run ESLint
+        run: |
+          npx eslint src --ext .js,.jsx
```
Comment (schatten) [inline .github/workflows/hma-webapp-ci.yaml:29]: <3

File: hasher-matcher-actioner/webapp/package.json [hunk 3]
```diff
@@ -5,8 +5,12 @@
   "dependencies": {
     "@testing-library/jest-dom": "^5.11.9",
     "@testing-library/react": "^11.2.5",
-    "@testing-library/user-event": "^12.8.1",
+    "@testing-library/user-event": "^13.0.1",
     "bootstrap": "^4.6.0",
+    "eslint": "^7.22.0",
```
Comment (schatten) [inline hasher-matcher-actioner/webapp/package.json:10]: nit: Is it customary for things to be dependencies AND devDependencies?

While it is unlikely someone does `npm install <hma>`, this seems weird.

**Review by schatten — APPROVED** [review]
This looks great!

Great that we renamed the files as `.jsx`. It's more indicative, perhaps vscode will behave better now.

This is good work. Linters and general tooling is hard to miss these in the first flurry of pull requests, but I'm glad you took the time! This provides necessary guard rails. :) 



---
nit: When you are doing rename and loads of changes, best to do `git mv`. That way git recognizes that you are renaming and doesn't show full delete + full add instead.

AFAIK, there is no way to make this happen after you've already renamed. Git sucks in some ways.

File: hasher-matcher-actioner/webapp/package.json (hunk 3 above)
Comment (BarrettOlson) [reply hasher-matcher-actioner/webapp/package.json]: good catch, these don't need to be in dependencies; moved them to devDependencies.

**Review by eriksendc — APPROVED** [review]

**Pull request merged**